test/test_nn.py

```diff
@@ -3809,10 +3810,18 @@ def get_modules():
 
                 # TEST TRAINING BEHAVIOR
 
-                # run forward again and assert that u and v are updated
+                # We perform GD first to modify the initial matrix
+                wrapped_m(input).sum().backward()
+                with torch.no_grad():
+                    for p in wrapped_m.parameters():
+                        if p.requires_grad:
+                            p.add_(- p.grad, alpha=0.01)
+
                 out = wrapped_m(input)
-                self.assertNotEqual(u0, spectral_norm_m.u)
-                self.assertNotEqual(v0, spectral_norm_m.v)
+                if requires_grad:
+                    # run forward again and assert that u and v are updated
+                    self.assertNotEqual(u0, spectral_norm_m.u)
```

Comment:
I'd probably agree with @albanD here, with having `u` and `v` be `_u` and `_v` instead. Even though `_SpectralNorm` is private, `m.parametrizations.weight[0]` seems to be a public api no?